src/android/toga_android/app.py

```diff
@@ -92,3 +112,19 @@ def set_on_exit(self, value):
 
     def add_background_task(self, handler):
         self.loop.call_soon(wrapped_handler(self, handler), self)
+
+    async def invoke_intent_for_result(self, intent):
+        """
+        Calls an Intent and waits for its result
+
+        :param Intent intent: The Intent to call
+        :returns: A Dictionary containing "resultCode" (int) and "resultData" (Intent or None)
+        :rtype: dict
+        """
+        self._listener.last_intent_requestcode += 1
+        code = self._listener.last_intent_requestcode
+        result_future = asyncio.Future()
+        self._listener.running_intents[str(code)] = result_future
+        self.native.startActivityForResult(intent, code)
+        await result_future
```

Comment:
I like to do `return await result_future` as one line, rather than two lines like you do here on 129-130. Your way is fine, too, so not a blocker, but just something that occurred to me.

Reply:
Well, I do not return result_future, I need to return result_future.result()
And return await result_future.result() did not work.